### DISCUSSION TITLE
Analyzing network delays

### DISCUSSION THREAD

**Discussion by mnour12** [hofbi/telecarla#68] [Q&A]
Hi, thanks for the efforts on this framework.

Using tc-netem, I wanted to test the impact of different network delay profiles on the ToD operation. So, I wanted to collect information about the network. I initially used wireshark to capture the RTP/RTSP packets on the client and server sides and compare the delay between them. But it seems that wireshark captures packets after the delay from tc-netem has been  introduced so it's not seen on the server side? Even though the impact is seen at the client side (video quality degradation, lagging...etc)

Is there another way to be able to calculate the actual G2G delay?

**Comment by cbachhuber** [comment]
In case you want to measure the actual G2G delay, you can use https://github.com/cbachhuber/G2GDelay. This requires [a little bit of hardware](https://github.com/cbachhuber/G2GDelay/blob/57c546cb53987db089dadcfdd840e663ccd23c40/README.md#construction-manual), though.

Disclaimer: I'm the creator of https://github.com/cbachhuber/G2GDelay.

**Comment by hofbi** [comment]
If you want to measure the G2G delay, @cbachhuber's approach seems reasonable.

**Comment by mnour12** [comment]
Thank you for the suggestion. Unfortunately, it's not feasible to use this approach right now because of the hardware requirements. 
Is there another way to possibly log the packets being sent/received using the RTSP server/client in telecarla?

**Comment by hofbi** [comment]
It is already some years ago when I wrote this, so cannot remember how I did it back them. Since you cannot really measure the G2G delay without any hardware, you could measure the individual components such as encoding, transmission, decoding, etc.
